### ISSUE TITLE
Summary of stats for css modules usage

### ISSUE THREAD

**Issue by export-mike** [postcss/postcss-devtools#18]
I'm using css-modules and from css-loader so this loads up multiple css files for a given build I wonder if theres a way to get an aggregate report on the results for a build? as I'm seeing metrics on different postcss plugins per css file.

I have too many to individually count through.

Ideally the same output but the sum of all stylesheets.

**Comment by ben-eb** [comment]
Pull requests welcome. 👍

**Comment by ai** [comment]
@ben-eb what API do you want?

Maybe:

``` js
let devtools = require('postcss-devtools')({ summary: true })

postcss([devtools, ...plugins]).process(css1)
postcss([devtools, ...plugins]).process(css2)
postcss([devtools, ...plugins]).process(css3)

devtools.printSummary()
```

**Comment by ben-eb** [comment]
`summary` looks good to me.

**Comment by export-mike** [comment]
@ben-eb @ai where in the process would this fit in? I'm happy to help out

**Comment by ai** [comment]
@export-mike you could send PR.

Just don’t print statistics on `summary: true` option and add addition `printSummary()` method to returned plugin.

**Comment by export-mike** [comment]
(updated) Ah this makes sense Ignore previous comment. we only print when devtools.printSummary() is called?

**Comment by export-mike** [comment]
if we print when `summary: true` how can we hook into the end of the pipeline to output the results? is there a callback for this?

**Comment by ai** [comment]
You could add some extra plugin to the end of plugins list dynamically:

```
let summarize = function (css, result) {
  findAndRemoveSelf(result.processor.plugins)
}

postcss.plugin('postcss-devtools', () => {
  return (css, result) => {
    result.processor.plugins.push(summarize)
  }
})
```

Don’t forget to remove this subplugin in its body

**Comment by export-mike** [comment]
thanks that makes sense, so I think we may need the following _default_ options:

```
summaryPerFile: true 
summary: false
```

summaryPerFile: being the existing behaviour. Users with lots of css files are likely to not want to see this existing output as its just too much.

summary: being the new summary at the end of the pipeline.

users can have both flags enabled and can view existing output and the total summary for all css files in the pipeline.

@ai removing the subplugin we create is just to reduce any side-affects it may cause?

**Comment by ai** [comment]
@export-mike stop-stop :) end of `process()` doesn’t mean end of evey file :)

For example gulp-postcss calls `process()` for every single input file :).

So you need some `devtool.printSummary()` call anyway :(

**Comment by export-mike** [comment]
I was hoping this isn't the case. 

Thanks for clarifying

Users of gulp, webpack etc will need to know how to hook into the end of their build process to call printSummary().  I've found a way to do this in webpack. https://github.com/webpack/webpack-dev-server/issues/132

**Comment by ben-eb** [comment]
Released in 1.1.0.

https://github.com/postcss/postcss-devtools/commit/65da9f6ab515217318b2961a2dee6872e97958b6
